ERCS/erc-7506.md

```diff
@@ -0,0 +1,463 @@
+nature. While the decentralization movement has brought about significant improvements around trustless technologies,
+many ecosystems building on top of these are in need of technologies that build trust in their realm. Real-world
+applications have shown that verifiable claims alone are not enough for this purpose. Moreover, a supporting layer of
+on-chain metadata is needed to support a reliable exchange and verification of those claims.
+
+The absence of a structured mechanism to manage these claim metadata on-chain poses a significant hurdle to the
+formation and maintenance of trust among participating entities in an ecosystem. This necessitates the introduction of a
+layer of on-chain metadata, which can assist in the reliable verification and interpretation of these claims. Termed "
+hints" in this specification, this metadata can be used in numerous ways, each serving to bolster the integrity and
+reliability of the ecosystem's claims. Hints can perform various tasks, such as providing revocation details,
+identifying trusted issuers, or offering timestamping hashes. These are just a few examples that enable ecosystems to
+validate and authenticate claims, as well as verify data integrity over time.
+
+The proposed "Trusted Hint Registry" aims to provide a robust, flexible, and standardized interface for managing such
+hints. The registry allows any address to manage multiple lists of hints, with a set of features that not only make it
+easier to create and manage these hints but also offer the flexibility of delegating these capabilities to trusted
+entities. In practice, this turns the hint lists into dynamic tools adaptable to varying requirements and use cases.
+Moreover, an interface has been designed with a keen focus on interoperability, taking into consideration existing W3C
+specifications around Decentralized Identifiers and Verifiable Credentials, as well as aligning with on-chain projects
+like the Ethereum Attestation Service.
+
+By providing a standardized smart contract interface for hint management, this specification plays an integral role in
+enabling and scaling trust in decentralized ecosystems. It offers a foundational layer upon which claims—both on-chain
+and off-chain—can be reliably issued, verified, and interpreted, thus serving as an essential building block for the
+credible operation of any decentralized ecosystem. Therefore, the Trusted Hint Registry is not just an addition to the
+ecosystem but a necessary evolution in the complex topology of decentralized trust.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and
+“OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+This EIP specifies a contract called`TrustedHintRegistry`that is deployed once and may then be commonly used by a wider
+audience. Governance is deliberately excluded from this ERC.
```

Comment:
If there is only a single registry contract deployed, does there need to be a standard? EIPs/ERCs generally shine when multiple independent implementers need to interoperate without directly communicating. ERC-20 is a prime example: wallets shouldn't have to write bespoke code for each token, and each token wants to work with as many wallets as possible. Having a token standard makes sense.

In this proposal, you define a single registry contract. Once it's deployed, anyone can use its public interface, no standard required.

That said, perhaps I'm misinterpreting this statement? If there's only one instance ever deployed, I'm not sure why any of the functions defined later on would ever not be implemented. Several (like `getMetadata`) seem to be optional.

---

We have made exceptions to this rule for some ERCs (notably [ERC-7409](https://eips.ethereum.org/EIPS/eip-7409).) If you'd like to go down that route, I'd suggest specifying the same deployment methodology as ERC-7409.

Reply:
Great question as this section definitely needs some work to be more clear. Let me try to explain it in more detail.

TL;DR: We should remove the sentence as the standard should not care about it. Multiple deployments will probably still be needed to reflect ecosystems needs. Standardizing the contract makes it possible to have a common interface, e.g., for those general checks needed to do while verifying claims across ecosystems.

One area where this ERC might be helpful is, e.g., digital identities (see, e.g.,off-chain standards like W3C Decentralized Identifiers and Verifiable Credentials or your on-chain addresses). In general, those digital identities can represent some on/off-chain entity and can issue signed claims to other entities, receive them, or present them to proof something to some other entity.

In reality, there are often ecosystem-specific claims that are being issued by specific organizations vetted trusted in their respected ecosystems. Think of KYC for DeFi, where me issuing a KYC claim attesting your KYC status is worth nothing, but Coinbase doing it might be trusted in a DeFi ecosystem. To keep track of those trusted entities/ identities, the ecosystem can come together and publish them as trusted hints in this proposed trusted hint registry. The sentence `[...] deployed once and may then be commonly used by a wider
audience. [...]` is supposed to reflect that many entities can be part of that specific ecosystem and use the one contract to publish trusted hints in the same namespace/list combo or separated in different namespace/list combos. So one group might be allowed to define trusted entities for KYC claims, another for AML claims, and so on.

The idea is, that every ecosystem might have their own deployment of this registry to publish trusted hints, that consuming parties, e.g. during verification of claims, can consume. Some methods are marked optional because they might be overkill for some ecosystems but present a standardized way on list/ hint management features in case they are needed. The core methods about setting hints are required, while nice-to-have management features (like adding delegates, meta txn, ...) are kept optional to not overcomplicate things in case an ecosystem might not need it. Ecosystems will probably also have the need for use-case specific methods that would warrant their own deployment.

I know the example above with trusted entities is very specific, but the idea of trusted hints can be expanded further, as they can represent revocation information about issued claims, timestamping for re-verifiability for off-chain claims, and more. 

At the company I work with, we heavily use W3C Decentralized Identifiers and Verifiable Credentials for corporate identities and signed claims that represent, e.g., an enterprises authorized trading partner status needed to legally exchange drugs in your supply chain according to the US [DSCSA](https://www.fda.gov/drugs/drug-supply-chain-integrity/drug-supply-chain-security-act-dscsa). In this W3C off-chain claim world, the identifier part is already solved with the did:ethr method based on [ERC-1056](https://eips.ethereum.org/EIPS/eip-1056). Though, a lot of the other core problems around trusted issuers and revocation are not solved yet with a resilient, decentralized architecture. This ERC is supposed to close this gap, while being, imo, relevant to the core Ethereum community for on-chain claims. We were already approached by other consortiums and their ecosystems to talk about this hint registry because it solves a lot of problems every ecosystem faces. This standard makes it possible to define a common language on how to set and retrieve this information across ecosystems and deployments on Ethereum.

We should probably remove that sentence because there is no good reason that the standards defines this, as this questions would have to be answered on another layer. I will incorporate fixes to your other feedback in a commit later.

Reply:
Perfect, thanks a lot for taking the time to write that out! I think you've convinced me that this is a good fit for an EIP.

---

As an aside, you can probably take some of this comment and add it to the EIP itself (likely motivation.)